sklearn/inspection/_plot_estimators.py

```diff
@@ -0,0 +1,252 @@
+}
+.sk-parallel-item::after {
+  content: "";
+  width: 100%;
+  border-bottom: 1px solid gray;
+  flex-grow: 1;
+}
+.sk-serial::before {
+  content: "";
+  position: absolute;
+  border-left: 1px solid gray;
+  top: 2em;
+  bottom: 0;
+  left: 50%;
+}
+.sk-serial {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  float: left;
+  background: white;
+}
+.sk-parallel {
+  display: flex;
+  align-items: stretch;
+}
+.sk-parallel-item {
+  display: flex;
+  flex-direction: column;
+  position: relative;
+}
+.sk-parallel-item:first-child::after {
+  align-self: flex-end;
+  width: 50%;
+}
+.sk-parallel-item:last-child::after {
+  align-self: flex-start;
+  width: 50%;
+}
+.sk-final-spacer {
+  visibility: hidden;
+  font-family: monospace;
+  white-space: pre;
+}
+.sk-dashed-wrapped {
+  border: 1px dashed gray;
+  padding: 0.25em;
+}
+.sk-label {
+  text-align: center;
+  font-family: monospace;
+  font-weight: bold;
+  margin: 0;
+  background: white;
+}
+.sk-serial-item {
+  margin-bottom: 0.25em;
+}
+.sk-container {
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+}
+.sk-container > .sk-serial::before {
+  border-left: 0;
+}
+[sk-data-tooltip] {
+  position: relative;
+  cursor: pointer;
+}
+[sk-data-tooltip]:before {
+  visibility: hidden;
+  opacity: 0;
+  pointer-events: none;
+  font-weight: 400;
+}
+[sk-data-tooltip]:before {
+  position: absolute;
+  top: 0;
+  left: 0;
+  padding: 0.5em;
+  overflow: hidden;
+  background-color: #f0f8ff;
+  border: 1px solid gray;
+  white-space: pre;
+  content: attr(sk-data-tooltip);
+  text-align: left;
+}
+[sk-data-tooltip]:hover:before {
+  visibility: visible;
+  opacity: 1;
+  z-index: 2;
+}
+"""
+
+
+def export_html(estimator, print_changed_only=True):
+    """Build a html representation of an estimator
+
+    Parameters
+    ----------
+    estimator : estimator object
+        The estimator to visualize.
+
+    print_changed_only : bool, optional (default=True)
+        If True, only the parameters that were set to non-default
+        values will be printed when printing an estimator.
+    """
+
+    with config_context(print_changed_only=print_changed_only), \
+            closing(StringIO()) as out:
+
+        if not isinstance(estimator, Pipeline):
+            estimator = Pipeline([('', estimator)])
+
+        out.write('<html><head><style>')
+        out.write(_STYLE.replace('\n', ''))
+        out.write('</style></head><body>')
+
+        out.write('<div class="sk-container">')
+        _write_estimator_html(out, estimator, '')
+        out.write('</div>')  # sk-container
+
+        # Adds whitespace at the end to allow space for hover info
+        out.write('<div class="sk-final-spacer">')
+        out.write(_estimator_tool_tip(estimator.steps[-1]))
+        out.write('</div>')  # sk-final-spacer
+        out.write("</body></html>")
+
+        html_output = out.getvalue()
+
+        try:
+            from IPython.display import HTML
```

Comment:
Hmm... I think we should more specifically determine whether we are in an ipython evaluation context, or at a minimum, check if 'IPython' is in sys.modules